app/views/video/home.html.haml

```diff
@@ -3,7 +3,7 @@
     %header
       %h3= link_to "#{category.name}", show_category_path(category)
     .videos.row
-      - category.videos.each do |video|
+      - category.recent_videos.each do |video|
         .video.col-sm-2
-          = link_to show_video_path(video.id.to_s) do
+          = link_to video_path(video.id.to_s) do
```

Comment:
You can just do `video_path(video)`, you could also do `.id`, but you don't need the `to_s`. The data will be passed as a string in the http request automatically. The reason `video` works by itself is because of the `to_param` method. http://apidock.com/rails/ActiveRecord/Base/to_param